mm2src/coins/tendermint/tendermint_balance_events.rs

```diff
@@ -1,27 +1,45 @@
 use async_trait::async_trait;
-use common::{executor::{AbortSettings, SpawnAbortable},
-             http_uri_to_ws_address, log, PROXY_REQUEST_EXPIRATION_SEC};
-use futures::channel::oneshot::{self, Receiver, Sender};
+use common::{http_uri_to_ws_address, log, PROXY_REQUEST_EXPIRATION_SEC};
+use futures::channel::oneshot;
 use futures_util::{SinkExt, StreamExt};
 use jsonrpc_core::{Id as RpcId, Params as RpcParams, Value as RpcValue, Version as RpcVersion};
-use mm2_core::mm_ctx::MmArc;
-use mm2_event_stream::{behaviour::{EventBehaviour, EventInitStatus},
-                       ErrorEventName, Event, EventName, EventStreamConfiguration};
+use mm2_event_stream::{Broadcaster, Event, EventStreamer, NoDataIn, StreamHandlerInput};
 use mm2_net::p2p::Keypair;
 use mm2_number::BigDecimal;
 use proxy_signature::RawMessage;
 use std::collections::{HashMap, HashSet};
 
 use super::TendermintCoin;
-use crate::{tendermint::TendermintCommons, utxo::utxo_common::big_decimal_from_sat_unsigned, MarketCoinOps, MmCoin};
+use crate::{tendermint::TendermintCommons, utxo::utxo_common::big_decimal_from_sat_unsigned, MarketCoinOps};
 
-#[async_trait]
-impl EventBehaviour for TendermintCoin {
-    fn event_name() -> EventName { EventName::CoinBalance }
+pub struct TendermintBalanceEventStreamer {
+    coin: TendermintCoin,
+}
+
+impl TendermintBalanceEventStreamer {
+    pub fn new(coin: TendermintCoin) -> Self { Self { coin } }
+}
 
-    fn error_event_name() -> ErrorEventName { ErrorEventName::CoinBalanceError }
+#[async_trait]
+impl EventStreamer for TendermintBalanceEventStreamer {
+    type DataInType = NoDataIn;
+
+    fn streamer_id(&self) -> String { format!("BALANCE:{}", self.coin.ticker()) }
```

Comment:
It's probably better to create strict type and functions for generating event types (e.g., EventType enumeration which implements `ToString`, which can be used as `EventType::Balance(ticker).into()/to_string()`) instead of building strings in every event implementation.

Reply:
> (e.g., EventType enumeration which implements ToString, which can be used as EventType::Balance(ticker).into()/to_string())

tbh i didn't want to have some enum in `mm2_event_streaming` that refers to specific events in mm2 (so the crate is as general as possible). we can try trait + simple struct for each event type.

Reply:
> tbh i didn't want to have some enum in mm2_event_streaming that refers to specific events in mm2 (so the crate is as general as possible).

You could create it in different module.